tools/statplot.py

```diff
@@ -183,51 +189,62 @@ def KeyPressed(self, widget, event):
                 or self.yCombo.get_child().has_focus()):
             pass
         elif key == 'r':
-            self.RefreshData()
+            self.entries, self.values = self.ReadData(self.statfile)
             self.PlotData('update', self.PlotType, self.ax.get_xscale(),
                           self.ax.get_yscale())
         elif key == 'q':
             self.destroy()
         elif key == 'x' or key == 'y':
-            if key == 'x':
-                self.get_scale = self.ax.get_xscale
-                curData = self.xData
-            elif key == 'y':
-                self.get_scale = self.ax.get_yscale
-                curData = self.yData
-            if (self.get_scale() == 'linear'
-                    and (curData == 0).all()):
-                warnings.warn('Change to logarithmic scale denied: the '
-                              + 'selected variable for the ' + key + ' axis '
-                              + 'is null.', stacklevel=2)
-                scale = 'linear'
-            elif (self.get_scale() == 'linear'
-                    and max(abs(curData)) / min(abs(curData)) < 10):
-                warnings.warn('Change to logarithmic scale denied: the '
-                              + 'selected variable for the ' + key + ' axis '
-                              + 'has a range of variation smaller than one '
-                              + 'order of magnitude.', stacklevel=2)
-                scale = 'linear'
-            elif self.get_scale() == 'linear':
-                axMin = min(abs(curData[curData != 0]))
-                axMax = max(abs(curData))
-                if axMin != axMax and min(curData) < 0:
-                    scale = 'symlog'
-                elif axMin != axMax:
-                    scale = 'log'
-                else:
+            if self.values.ndim == 1:
+                warnings.warn('Insufficient data available to turn on '
+                              + 'logarithmic scale', stacklevel=2)
+            else:
```

Comment:
You could get rid of a level of indentation here with an early return (de-indent the `else` block). Also, fun fact: you don't actually need to explicitly add the string parts together. I think it looks cleaner without, but I'll defer to your judgement!

```suggestion
            if self.values.ndim == 1:
                warnings.warn('Insufficient data available to turn on '
                              'logarithmic scale', stacklevel=2)
                return
```